Updated Auth Flow

Mnpn opened the pull request:
[trello/74](https://trello.com/c/iLEFbtyu/74-popup-on-incorrect-login-details), [trello/68](https://trello.com/c/96yZPJ7C/68-design-update-sign-in-flow) - This PR finishes the authentication UI, with the landing page and sign up/log in page tweaks from Figma.

There is a working language toggle button, which currently just prints to console but let's see if we ever hook that up to something.

If wrong credentials are entered, a popup will now appear.

<img src="https://github.com/M-AboWarda/AgeWell-Meals/assets/17569187/dd841992-c677-4fdc-aeb8-fa169acc638f" width="50%">
<img src="https://github.com/M-AboWarda/AgeWell-Meals/assets/17569187/172dadc4-deb5-464c-b938-fe49bdb456ac" width="50%">
<img src="https://github.com/M-AboWarda/AgeWell-Meals/assets/17569187/e850b39f-b42e-4ccb-b6b5-13aee71652f3" width="50%">
<img src="https://github.com/M-AboWarda/AgeWell-Meals/assets/17569187/c6f4b8da-387b-411c-a05c-07daddddc3a1" width="50%">

## Before this can be merged
This PR depends on other PRs - please review and merge the following so I can sort out conflicts and undraft
- The white logo I wish to use on the landing page instead of the current green depends on #19.
- An icon in the error popup depends on #18.

nouradeen reviewed (changes requested):
## **login.dart**
The `SafeArea()` should be wrapped with a `SingleChildScrollView()` to overcome the problem of overflow when the keyboard is opened. Otherwise, the keyboard will block the view of the input field and we get an overflow.

## **sign_up.dart**
Same thing as **login.dart**.

All fields in the signup should be mandatory, as the customer wants all the information, you can achieve that by adding the option `validator` like this, which checks the value in the input field:
``` dart
InputTextField(
                  labelText: 'Full Name',
                  suffixIcon: const Icon(Icons.person),
                  controller: nameController,
                  validator: (value) {
                    if (value!.isEmpty) {
                      return "Enter Name";
                    }
                    return null;
                  },
                ),
```

- line 253 `Navigator.push()`: We don't want to keep pushing stuff in the stack. We want to keep the stack as empty as possible. If we just push, the back button will be useless because its only job is to pop screens from the stack.

## Comments
**this is, quite frankly, retarded - "input"textfields cant have backgrounds apparently** *in sign_up.dart line 80*: 
This is my own custom input field :), if you want to add a background color, just modify the `input_field.dart` under the widgets folder, and add a parameter in the class.

Mnpn commented:
Sorry in advance for this long post
- Got it - sorry, I didn't test it with an on-device keyboard. They now have a `SingleChildScrollView()`, and appear to work after some other wrapping to make it be positioned right (it otherwise just refused to render because of a Spacer), I also removed the resizeToAvoidBottomInset, because otherwise it can't scroll and you can't see the text field you're typing in (keyboard covers it).. This was a bit of a pain and is probably not perfect but it appears to work correctly and scroll, with the background still attached to the back..
- It looks like you've already made the fields mandatory - I've merged main back into here (to resolve the conflict) and that should have fixed that.

<img src="https://github.com/M-AboWarda/AgeWell-Meals/assets/17569187/b383d218-5f41-4fca-ab60-dfb320704aba" width="30%">

- Added background option to InputTextField
    - Speaking of which, InputTextField is now used on the sign in page too
- Regarding Navigator.push, good catch. This new flow doesn't have a back arrow here, but I assume the back button on an Android device still pops back a step. Once you're past the landing page, you arrive at the login page, and get to go to the sign up if you want. It's probably still better practice to just step back a step when you've tapped to go back to the sign in page, so I have done that now with a `Nagivator.pop()`, but now if you go directly to sign up and hit "sign in", it'll bring you back to the landing page of course.. not quite sure how to solve this other than to either check where you came from, or just keep stacking with new pushes? Aaaaa, what do you think we should do? I kinda don't want to spend more time on this to be honest, when I could be working on other screens, so maybe we should just leave it be for now

nouradeen reviewed (approved):
- Don't worry about the push, pop, replace, etc... I had it working on the first commit of the project 😄 , I can fix it later

**LGTM**
